pkg/reward/reward.go

```diff
@@ -4,44 +4,66 @@
 package reward
 
 import (
+	"context"
 	"fmt"
+	"math"
 	"math/big"
+	"strings"
 	"time"
 
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi"
 	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
 	"github.com/go-kit/kit/log"
 	"github.com/go-kit/kit/log/level"
 	"github.com/pkg/errors"
 	"github.com/tellor-io/telliot/pkg/aggregator"
-	"github.com/tellor-io/telliot/pkg/ethereum"
+	"github.com/tellor-io/telliot/pkg/contracts"
+	"github.com/tellor-io/telliot/pkg/contracts/tellor"
+	eth "github.com/tellor-io/telliot/pkg/ethereum"
 )
 
 const ComponentName = "reward"
 
 type ContractCaller interface {
 	GetUintVar(opts *bind.CallOpts, _data [32]byte) (*big.Int, error)
 	CurrentReward(opts *bind.CallOpts) (*big.Int, error)
+	GetNewCurrentVariables(opts *bind.CallOpts) (struct {
+		Challenge  [32]byte
+		RequestIds [5]*big.Int
+		Difficutly *big.Int
+		Tip        *big.Int
+	}, error)
 }
 
-func New(logger log.Logger, aggr aggregator.IAggregator, contractCaller ContractCaller) *Reward {
+func New(logger log.Logger, aggr aggregator.IAggregator, contractCaller ContractCaller, client contracts.ETHClient, account *eth.Account) *Reward {
+	// Getting abi.
+	abi, _ := abi.JSON(strings.NewReader(tellor.TellorABI))
 	return &Reward{
 		aggr:           aggr,
 		logger:         log.With(logger, "component", ComponentName),
 		contractCaller: contractCaller,
 		gasUsed:        make(map[int64]*big.Int),
+		client:         client,
+		abi:            abi,
+		account:        account,
```

Comment:
I think instead of adding it here this should be added as an argument to `GasUsed`
`GasUsed(....,account)`